app/Connect/RouteServiceProvider.php

```diff
@@ -24,87 +23,39 @@ public function map(): void
     }
 
-    private function legacyCompatRoutes(): void
+    private function rpcConnectRoutes(): void
     {
         /*
-         * Backwards compatibility for "rpc" api client integrations.
-         * The new APIs split Web App from RPC from API and are hosted on their own subdomain each.
-         * Legacy endpoints are served from web app domain :/ -> lock into appDomain().
-         * We don't want legacy endpoints on the dedicated subdomain or vice versa.
-         * Otherwise we'd have to maintain/support them there, too - have a clean cut instead
-         * Force API/RPC context for these routes by allowing Route::any() methods.
-         * Otherwise the web app may respond with html.
-         * Check method and fail within action instead to make sure it will be a json response.
-         * E.g. app_login.php never listened for GET requests -> throw a MethodNotAllowedException
+         * Connect RPC API routes for client integrations (RAIntegration, RetroArch).
+         * These routes use the HandlesPublicFileRequests trait to load public PHP files.
+         *
+         * Route::any() is used because different clients use different HTTP methods.
+         * RAIntegration uses POST requests, except for LatestIntegration.html.
+         * Legacy RetroArch uses GET requests. RetroArch 1.9.13+ (Nov 2021) should use POST.
          *
          * Note: Don't apply 'auth:connect-token' guard via middleware for the whole of the RPC API.
          * There are public routes in there.
-         * Authorize in controllers where needed; Auth::shouldUse('connect-token'); is applied.
+         *
+         * TODO: Eventually migrate to a proper controller-based architecture.
+         * Previously attempted with ConnectApiController and related Concerns
+         * (AchievementRequests, BootstrapRequests, HeartbeatRequests, LeaderboardRequests,
+         * LegacyCompatProxyRequests), but removed as it wasn't the right direction at this time.
+         * Future implementation should follow standard Laravel MVC patterns and ideally host
+         * the Connect API on a dedicated subdomain (see connectDomain/connectPrefix methods below)
+         * to cleanly separate RPC traffic from the main web application. This work was previously
+         * started in the site's V2 release (circa 2022), but ultimately removed because PHPStan was
+         * complaining about dead code.
+         * @see https://github.com/RetroAchievements/RAWeb/blob/d81dfbfd06d3233f73168546467e3e6c8006d124/app/Connect/Controllers/ConnectApiController.php
          */
         Route::middleware(['connect'])->group(function () {
-            /*
-             * RAIntegration uses POST requests, except for LatestIntegration.html
-             * Legacy RetroArch uses GET requests except, RetroArch 1.9.13+ (Nov 2021) should be using POST for everything
-             */
             Route::any('login_app.php', fn () => $this->handleRequest('login_app'));
```

Comment:
This file no longer exists: https://github.com/RetroAchievements/RAWeb/pull/4027#discussion_r2436548388